components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/util/HandlerManagerTest.java

```diff
@@ -0,0 +1,122 @@
+        List<RegistrationHandler> registrationHandlers3 = new ArrayList<>();
+        return new Object[][] {
+                {dcrMessageContext, null},
+                {dcrMessageContext, registrationHandlers1},
+                {dcrMessageContext, registrationHandlers2},
+                {dcrMessageContext, registrationHandlers3}
+        };
+    }
+
+    @Test(dataProvider = "BuildRegistrationHandlers")
+    public void testGetRegistrationHandler(Object dcrMessageContext, Object handlers)  {
+        mockStatic(DCRDataHolder.class);
+        when(DCRDataHolder.getInstance()).thenReturn(dataHolder);
+        when(dataHolder.getRegistrationHandlerList()).thenReturn((List<RegistrationHandler>) handlers);
+        try {
+            Assert.assertNotNull(HandlerManager.getInstance()
+                    .getRegistrationHandler((DCRMessageContext) dcrMessageContext));
+        } catch (IdentityRuntimeException e) {
+            Assert.assertTrue(true);
+        }
+    }
+
+    @DataProvider(name = "BuildUnRegistrationHandlers")
+    public Object[][] BuildUnRegistrationHandlers() {
+        Map<String,String> param = new HashMap<String, String>();
+        param.put("client_id","N2QqQluzQuL5X6CtM3KZwqzLQhUa");
+        param.put("client_secret","4AXWrN88aEfMvq2h_G0dN05KRsUa");
+        DCRMessageContext dcrMessageContext = new DCRMessageContext(param);
+        UnRegistrationHandler unRegistrationHandler1 = new UnRegistrationHandler();
+        UnRegistrationHandler unRegistrationHandler2 = new UnRegistrationHandler();
+        List<UnRegistrationHandler> unRegistrationHandlers1 = new ArrayList<>();
+        unRegistrationHandlers1.add(unRegistrationHandler1);
+        unRegistrationHandlers1.add(unRegistrationHandler2);
+        List<UnRegistrationHandler> unRegistrationHandlers2 = new ArrayList<>();
+        unRegistrationHandlers2.add(unRegistrationHandler1);
+        List<UnRegistrationHandler> unRegistrationHandlers3 = new ArrayList<>();
+        return new Object[][] {
+                {dcrMessageContext, null},
+                {dcrMessageContext, unRegistrationHandlers1},
+                {dcrMessageContext, unRegistrationHandlers2},
+                {dcrMessageContext,unRegistrationHandlers3}
+        };
+    }
+
+    @Test(dataProvider = "BuildUnRegistrationHandlers")
+    public void testGetUnRegistrationHandlerException(Object dcrMessageContext, Object handlers){
+        mockStatic(DCRDataHolder.class);
+        when(DCRDataHolder.getInstance()).thenReturn(dataHolder);
+        when(dataHolder.getUnRegistrationHandlerList()).thenReturn((List<UnRegistrationHandler>) handlers);
+        try {
+            Assert.assertNotNull(HandlerManager.getInstance()
+                    .getUnRegistrationHandler((DCRMessageContext) dcrMessageContext));
+        } catch (IdentityRuntimeException e) {
+            Assert.assertTrue(true);
+        }
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
```

Comment:
Is this necessary since you have mentioned the object factory in testng.xml ?

Reply:
Yes. We need to have this or extend PowerMockTestCase